tests/Records/ValueObjects/Email.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crell\Serde\Records\ValueObjects;
+
+use Crell\Serde\Attributes\Field;
+
+class Email
+{
+    public function __construct(#[Field(serializedName: 'email')] public string $value) {}
```

Comment:
This way it (also applicable for `Age` ) forces properties to have a specific name, which depending on the context may be ambiguous, or even worse, it may be necessary to have, let's imagine, 2 email objects.

Ideally the property name would be dictated by the parent object, `People` in this case, or as a fallback, the `serializedName` to be infered there.

Another way to put it, is to consider this scenario:

```php
<?php

declare(strict_types=1);

namespace Crell\Serde\Records\ValueObjects;

use Crell\Serde\Attributes\Field;

class Invite
{
    public function __construct(
        #[Field(flatten: true)]
        public Email $invitingEmail,
        #[Field(flatten: true)]
        public Email $invitedEmail,
    ) {}
}
```

Forcing the `serializedName` on the value object makes this object impossible to deserialize/serialize, as far as I understand.

Does my perspective on this make sense?

Reply:
Hm, yes, it does break down if the value object is used twice.  However, I'm not really sure how to have an alternate name injected from the parent.

Part of the challenge is that there are certain approaches that would only work for a single-property value object, but there's nothing inherent in value objects that makes them single-value only.  (You could have a phone number class that stores the area code separately internally, for whatever reason.)  So any solution would need to handle that, too.

Which suggests, maybe using `renameWith: new Prefix(...)` on the parent class.  But then the question is how that interacts with the names in the value object.  I'm not sure what interaction there makes sense without some completely separate pathway.

Reply:
I'm delighted to see this PR.

But as already noted before, setting the field's name into the child object might be troublesome. You can't say how the value object will be used. 

For example, considering the type Age, a field of that type called `age` could exist, but also a `minimumAge` or `whatever`. Also, multiple fields could use the same type, so different names are needed.

It is the parent class, then, that should determine the name. 

What about a new attribute, as I put on the issue?

```php
#[Attribute(Attribute::TARGET_CLASS)]
readonly class ValueObject
{
    public function __construct(public string $field = 'value') {}
}
```
In this way, Serde knows how to serialize/hydrate. But this adds an extra dependency, and I prefer Vanilla PHP when possible.

As an alternative, why not use a flag like `simple`?

```php
readonly class Person
{
    public function __construct(
        #[Field(flatten: true, simple: true)]
        public Name $name,
        #[Field(flatten: true, simple: 'age')]
        public Age $age
    ) {}
}
```
I imagine two possible values: 
* `true` when the child value object uses the default field `value`
* `string` when the child uses a custom field name

Or better, why not merge `flatten` and `simple` behaviours in a single property called `simplify`?
```php
readonly class Person
{
    public function __construct(
        #[Field(simplify: true)]
        public Name $name,
        #[Field(simplify: 'age')]
        public Age $age
    ) {}
}
```
The field name or the `serializedName` might be used as a name.